### PR TITLE
change check_auth_status return type, add Clone trait

### PR DESCRIPTION
**Pull request by rdavis120**

### CODE REVIEW

File: .DS_Store
Comment (numberjuani) [inline .DS_Store:1]: add `.DS_Store` to `.gitignore`

File: src/models/market_data_history.rs [hunk 1]
```diff
@@ -0,0 +1,65 @@
+use chrono::naive::serde::ts_milliseconds;
+use chrono::NaiveDateTime;
+use serde::{Deserialize, Serialize};
+
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct Bar {
+    pub o: f64,
```
Comment (numberjuani) [inline src/models/market_data_history.rs:7]: `rust_decimal::Decimal` no `f64` please

File: src/models/market_data_history.rs [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct MarketDataHistory {
+    pub server_id: String,
+    pub symbol: String,
+    pub text: String,
+    pub price_factor: i64,
+    #[serde(with = "parse_datetime")]
+    pub start_time: NaiveDateTime,
+    #[serde(default)]
```
Comment (numberjuani) [inline src/models/market_data_history.rs:25]: you already impl `Default` for the whole struct, `#[serde(default)]` is not doing anything above these, also shouldn't the high and low be `rust_decimal::Decimal`?

File: src/models/market_data_history.rs [hunk 3]
```diff
@@ -0,0 +1,65 @@
+use chrono::naive::serde::ts_milliseconds;
+use chrono::NaiveDateTime;
+use serde::{Deserialize, Serialize};
+
+#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct Bar {
+    pub o: f64,
+    pub c: f64,
+    pub h: f64,
```
Comment (numberjuani) [inline src/models/market_data_history.rs:9]: lets rename all these fields to their full name close, high, low, open, then use `#[serde(alias = "c")]`

File: src/models/market_data_history.rs [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    pub price_display_value: String,
+    pub negative_capable: bool,
+    pub message_version: i64,
+    pub data: Vec<Bar>,
+    #[serde(default)]
+    pub points: u32,
+    pub travel_time: u32,
+}
+
+mod parse_datetime {
```
Comment (numberjuani) [inline src/models/market_data_history.rs:46]: we might look into moving this into a `utils` mod, looks familiar did you grab it from somewhere else? if so, then  we shouldn't write the same code twice, we should import it.

File: src/models/tickle.rs [hunk 5]
```diff
@@ -28,6 +28,7 @@ pub struct AuthStatus {
     pub connected: bool,
     pub message: String,
     pub server_info: ServerInfo,
+    pub fail: String,
```
Comment (numberjuani) [inline src/models/tickle.rs:31]: is this a new field?

File: src/rest/endpoints.rs [hunk 6]
```diff
@@ -29,16 +32,20 @@ impl IBClientPortal {
     }
     /// Current Authentication status to the Brokerage system.
     /// Market Data and Trading is not possible if not authenticated, e.g. authenticated shows false
-    pub async fn check_auth_status(&self) -> Result<Value, reqwest::Error> {
+    pub async fn check_auth_status(&self) -> Result<AuthStatus, reqwest::Error> {
```
Comment (numberjuani) [inline src/rest/endpoints.rs:35]: 👍

**Review by numberjuani — CHANGES_REQUESTED** [review]

File: src/models/tickle.rs (hunk 5 above)
Comment (rdavis120) [reply src/models/tickle.rs:31]: I only see fail on the `auth/status` endpoint output not on the /tickle -iserver - authStatus endpoint.  We could create a second AuthStatus struct if you would prefer that would include that field.

File: src/models/market_data_history.rs (hunk 4 above)
Comment (rdavis120) [reply src/models/market_data_history.rs:46]: I derived it from your `parse_date()` function used in `futures_contract.rs`.  They should probably be handled similarly.

File: src/models/market_data_history.rs (hunk 2 above)
Comment (rdavis120) [reply src/models/market_data_history.rs:25]: No high and low appear as strings like "123/456/789". I think we could derive a separate struct for them later.

```
string
High value during this time series with format %h/%v/%t. 
%h is the high price (scaled by priceFactor), 
%v is volume (volume factor will always be 100 (reported volume = actual volume/100)) and 
%t is minutes from start time of the chart
```

**Review by numberjuani — DISMISSED** [review]

**Comment by numberjuani** [comment]
.gitignore has a conflict, I think you can discard your change there, I applied it on my end. Then I'll approve and you can merge

**Pull request closed without merging**